e2e/nomostest/reset.go

```diff
@@ -478,34 +479,59 @@ func initialCommit(nt *NT, repoType gitproviders.RepoType, nn types.NamespacedNa
 	}
 }
 
-// TailReconcilerLogs starts tailing a reconciler's logs.
-// The logs are stored in memory until either the context is cancelled or the
-// kubectl command exits (usually because the container exited).
+// TailCSPodsAndReconcilerLogs starts tailing CS Pods and the reconciler's logs.
+// The pods and logs are stored in memory until either the context is cancelled
+// or the kubectl command exits (usually because the container exited).
 // This allows capturing logs even if the reconciler is deleted before the
 // test ends.
-// The logs will only be printed if the test has failed when the command exits.
-// Run in an goroutine to capture logs in the background while deleting RSyncs.
-func TailReconcilerLogs(ctx context.Context, nt *NT, reconcilerNN types.NamespacedName) {
-	out, err := nt.Shell.KubectlContext(ctx, "logs",
-		fmt.Sprintf("deployment/%s", reconcilerNN.Name),
-		"-n", reconcilerNN.Namespace,
-		"-c", reconcilermanager.Reconciler,
-		"-f")
-	nt.T.Logf("finished tailing logs for %s", reconcilerNN.String())
-	// Expect the logs to tail until the context is cancelled, or exit early if
-	// the reconciler container exited.
-	if err != nil && err.Error() != "signal: killed" {
-		// We're only using this for debugging, so don't trigger test failure.
-		nt.T.Logf("Failed to tail logs from reconciler %s: %v", reconcilerNN, err)
-	}
-	// When TailReconcilerLogs is called async, it's possible that the test fails
+// The pods and logs will only be printed if the test has failed when the
+// command exits.
+// Run in a goroutine to capture logs in the background while deleting RSyncs.
+func TailCSPodsAndReconcilerLogs(ctx context.Context, nt *NT, reconcilerNNs ...types.NamespacedName) {
+	var logs []string
+	// Tail CS pods
+	for _, ns := range CSNamespaces {
+		logs = append(logs, "kubectl get pods -n "+ns)
+		out, err := nt.Shell.Kubectl("get", "pods", "-n", ns)
+		if err != nil && err.Error() != "signal: killed" {
+			nt.T.Log("error running `kubectl get pods`:", err)
+		}
+		logs = append(logs, string(out))
+		logs = append(logs, "kubectl describe pods -n "+ns)
+		out, err = nt.Shell.Kubectl("describe", "pods", "-n", ns)
+		if err != nil && err.Error() != "signal: killed" {
+			nt.T.Log("error running `kubectl describe pods`:", err)
+		}
+		logs = append(logs, string(out))
+	}
+	nt.T.Log("finished tailing pods info")
+
+	// Tail reconciler logs
+	for _, reconcilerNN := range reconcilerNNs {
+		logs = append(logs, fmt.Sprintf("kubectl logs deployment/%s -n %s -c %s -f",
```

Comment:
suggest putting all the args in an array and joining them for the log. that way you don't need to duplicate building the command. you can do the same above too.

Reply:
SG

Reply:
Renamed `testPods` to `listAndDescribePods`